### PR TITLE
DEEPBLUE-285 - Send works to APTrust

### PR DESCRIPTION
**Pull request by fritzfreiheit**
https://mlit.atlassian.net/browse/DEEPBLUE-285

### CODE REVIEW

File: app/services/aptrust/aptrust_service.rb [hunk 1]
```diff
@@ -0,0 +1,350 @@
...
+
+    def attr_init( attr, work_attr )
+      attr ||= work_attr
+      if attr.blank?
+        attr = ''
+      else
+        attr = attr.squish[0..255]
+      end
+      return attr
+    end
```
Comment (ssciolla) [inline app/services/aptrust/aptrust_service.rb:45]: Trying to remember definition order rules in Ruby, can this be moved up?

File: app/services/aptrust/aptrust_service.rb [hunk 2]
```diff
@@ -0,0 +1,350 @@
...
+    def build
+      <<~INFO
+        Title: #{title}
+        Access: #{access}
+        Storage-Option: #{storage_option}
+        Description: #{description}
+        Item Description: #{item_description}
+        Creator/Author: #{creator}
+      INFO
+    end
```
Comment (ssciolla) [inline app/services/aptrust/aptrust_service.rb:56]: If we're following a similar serialization pattern for `bag-info.txt`, I wonder if storing the key names as strings with the value and then writing a generic serialization method writing out the key-value pairs would be appropriate. Just a thought.

File: app/services/aptrust/aptrust_service.rb [hunk 3]
```diff
@@ -0,0 +1,350 @@
...
+      # add bagit-info.txt file
+      # The length of the following 'internal_sender_description' does not work with the current bagit gem, maybe later.
+      # pub = monograph_presenter.publisher.blank? ? '' : monograph_presenter.publisher.first.squish[0..55]
+      # internal_sender_description = "This bag contains all of the data and metadata in a Monograph from #{pub} which has been exported from the Fulcrum publishing platform hosted at www.fulcrum.org."
+      {
+        BAG_INFO_KEY_SOURCE => BAG_INFO_VALUE_SOURCE,
+        BAG_INFO_KEY_COUNT => '1',
+        BAG_INFO_KEY_DATE => bag_date_now(),
+        'Internal-Sender-Description' => bag_description( work ),
+        'Internal-Sender-Identifier' => id_work( work )
```
Comment (ssciolla) [inline app/services/aptrust/aptrust_service.rb:173]: Might we want to add the "Source-Organization"? Should Bagging-Date just be the date (not the datetime), as per the instructions/spec? Datetime seems better but could be put elsewhere? https://aptrust.github.io/userguide/bagging/

File: app/services/aptrust/aptrust_service.rb [hunk 4]
```diff
@@ -0,0 +1,350 @@
...
+      "Bag for a #{work.class.name} hosted at deepblue.lib.umich.edu/data/" # TODO: improve this, or move to config
+    end
+
+    def self.bag_manifest( bag: )
+      bag.manifest!(algo: 'md5') # Create manifests
+
+      # HELIO-4380 demo.aptrust.org doesn't like this file for some reason, gives an ingest error:
+      # "Bag contains illegal tag manifest 'sha1'""
+      # APTrust only wants SHA256, or MD5, not SHA1.
+      # 'tagmanifest-sha1.txt' is a bagit gem default, so we need to remove it manually.
```
Comment (ssciolla) [inline app/services/aptrust/aptrust_service.rb:187]: So you can't specify SHA256 or MD5 for the tagmanifest?

File: app/services/aptrust/aptrust_service.rb [hunk 5]
```diff
@@ -0,0 +1,350 @@
...
+module Aptrust
+
+  # TODO: tracking of status of deposit
+
+  class AptrustInfo
+
+    mattr_accessor :aptrust_info_debug_verbose, default: true
+
+    # mattr_accessor :default_description, default: 'This bag contains all of the data and metadata related to a Monograph which has been exported from the Fulcrum publishing platform hosted at https://www.fulcrum.org. The data folder contains a Fulcrum manifest in the form of a CSV file named with the NOID assigned to this Monograph in the Fulcrum repository. This manifest is exported directly from Fulcrum's heliotrope application (https://github.com/mlibrary/heliotrope) and can be used for re-import as well. The first two rows contain column headers and human-readable field descriptions, respectively. {{ The final row contains descriptive metadata for the Monograph; other rows contain metadata for Resources, which may be components of the Monograph or material supplemental to it.}}'
+    mattr_accessor :default_description, default: 'The Description' # TODO
```
[the remark this answers is not in the document]
Comment (fritzfreiheit) [reply app/services/aptrust/aptrust_service.rb:12]: mattr_accessor and cattr_accessor are the same.

File: app/services/aptrust/aptrust_service.rb (hunk 3 above)
Comment (fritzfreiheit) [reply app/services/aptrust/aptrust_service.rb:173]: Yes, I think you're right.

File: app/services/aptrust/aptrust_service.rb (hunk 4 above)
Comment (fritzfreiheit) [reply app/services/aptrust/aptrust_service.rb:187]: Apparently not.

File: app/services/aptrust/aptrust_service.rb [hunk 6]
```diff
@@ -0,0 +1,350 @@
...
+    def initialize( work:,
+                    access: nil,
+                    creator: nil,
+                    description: nil,
+                    item_description: nil,
+                    storage_option: nil,
+                    title: nil )
+
+      @access = access; @access = attr_init( @access, 'Institution' )
+      @creator = creator; @creator = attr_init( @item_description, work.creator.join( " & " ) )
```
[the remark this answers is not in the document]
Comment (fritzfreiheit) [reply app/services/aptrust/aptrust_service.rb:30]: That's an error. Good catch.

**Pull request merged**